### PR TITLE
Update Swift code style guide to include some Swift 2.0 stuff

### PR DESCRIPTION
**Pull request by jvoll**
Guard is awesome, let's use it when possible!

@kschmidtdev, @jeremywiebe, and I agreed to leave error handling as is, at least for now, since we think it works better for our highly async code base. Feedback welcome!

### CODE REVIEW

File: swift/README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Component {
 How do I handle errors in Swift?
 ================================
 
-Swift doesn't have exceptions, so errors must somehow be returned to your caller.
+As of Swift 2.0 exceptions have been introduced into the language and many of the built-in APIs make use of them. We have decided to stick with our existing pattern of returning errors.
```
Comment (jeremywiebe) [inline swift/README.md]: Can we clarify that the APIs that use this are all the ones that used to use `someMethod(..., error: *NSError)` ?  People coming from Cocoa will be able to identify with that easily, I think.

File: swift/README.md [hunk 2]
```diff
@@ -249,6 +249,44 @@ if let message = message {
...
+```
+
+```swift
+// Good (Swift 2.0 way using guards)
+func myFunc(optionalKey: String?) {
+    guard let key = key else {
+        // log an error
+        return
+    }
+    guard let queue = dict[key] else { return }
```
Comment (jeremywiebe) [inline swift/README.md]: I know we're trying to be succinct here, but I think our code style also doesn't allow for these single-line blocks. :-)

File: swift/README.md [hunk 3]
```diff
@@ -249,6 +249,44 @@ if let message = message {
...
+    for queueMessage in queue {
+        guard let address = queueMessage["address"] as? MessageAddress else { continue }
```
Comment (jeremywiebe) [inline swift/README.md]: Oh, cool.  Didn't know you could use guard to short-circuit a loop!

File: swift/README.md (hunk 2 above)
Comment (jvoll) [inline swift/README.md]: Looking through it now I don't actually see us explicitly saying "no" to this. I thought you were right...but I don't see it. For empty closures we allow it to be on the same line but that's the closest thing I can see.

**Comment by kschmidtdev** [comment]
:+1:

File: swift/README.md [hunk 4]
```diff
@@ -249,6 +273,50 @@ if let message = message {
...
+```
+
+```swift
+// Good (Swift 2.0 way using guards)
+func myFunc(optionalKey: String?) {
+    guard let key = key else {
+        // log an error
+        return
+    }
+    guard let queue = dict[key] else {
```
Comment (jansepar) [inline swift/README.md]: Guard, so hot right now. Guard.

File: swift/README.md [hunk 5]
```diff
@@ -154,6 +154,30 @@ Place the closing `}` on the same line if the body of the block if empty:
...
+// bad
+guard let x = x else { return }
+
+// good
+guard let x = x else {
+    return
+}
+```
+
+
```
Comment (jeremywiebe) [inline swift/README.md]: pew pew

**Comment by jeremywiebe** [comment]
One extra line comment but :+1:

**Pull request merged**